prow/crier/controller.go

```diff
@@ -160,24 +175,40 @@ func (c *Controller) processNextItem() bool {
 			return true
 		}
 
-		if c.queue.NumRequeues(key) < 5 {
-			logrus.WithError(err).WithField("prowjob", keyRaw).Error("Failed processing item, retrying")
-			c.queue.AddRateLimited(key)
-		} else {
-			logrus.WithError(err).WithField("prowjob", keyRaw).Error("Failed processing item, no more retries")
-			c.queue.Forget(key)
-			utilruntime.HandleError(err)
+		return retry(c.queue, key, err)
+	}
+
+	// initiate PrevReportState, if not exist
+	if pj.Status.PrevReportState == nil {
+		pjCopy := pj.DeepCopy()
+		pjCopy.Status.PrevReportState = map[string]v1.ProwJobState{}
+		pjCopy.Status.PrevReportState[c.reporter.GetName()] = ""
+		if _, err := c.pjclientset.Prow().ProwJobs(pj.Namespace).Patch(pjCopy); err != nil {
```

Comment:
We run the risk here of nuking other fields that have changed, right? Need to patch only `status.prevReportState`?

Reply:
yeah I haven't figure out how to use Patch last night :-\

assume I only want to patch `Status.PrevReportState[c.reporter.GetName()]`, is that possible?

Reply:
Yes, there should be an API that accepts the `[]byte` explicit JSONPatch